packages/js/product-editor/src/blocks/generic/linked-product-list/edit.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import {
+	createElement,
+	useCallback,
+	useEffect,
+	useReducer,
+} from '@wordpress/element';
+import { useWooBlockProps } from '@woocommerce/block-templates';
+import { Product } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { ProductList } from '../../../components/product-list';
+import { ProductSelect } from '../../../components/product-select';
+import useProductEntityProp from '../../../hooks/use-product-entity-prop';
+import { ProductEditorBlockEditProps } from '../../../types';
+import {
+	getLoadLinkedProductsDispatcher,
+	getRemoveLinkedProductDispatcher,
+	getSearchProductsDispatcher,
+	getSelectSearchedProductDispatcher,
+	reducer,
+} from './reducer';
+import { LinkedProductListBlockAttributes } from './types';
+
+export function Edit( {
+	attributes,
+	context: { postType },
+}: ProductEditorBlockEditProps< LinkedProductListBlockAttributes > ) {
+	const { property } = attributes;
+	const blockProps = useWooBlockProps( attributes );
+	const [ state, dispatch ] = useReducer( reducer, {
+		linkedProducts: [],
+		searchedProducts: [],
+	} );
+	const loadLinkedProductsDispatcher =
+		getLoadLinkedProductsDispatcher( dispatch );
+	const searchProductsDispatcher = getSearchProductsDispatcher( dispatch );
+	const selectSearchedProductDispatcher =
+		getSelectSearchedProductDispatcher( dispatch );
+	const removeLinkedProductDispatcher =
+		getRemoveLinkedProductDispatcher( dispatch );
+
+	const [ linkedProductIds, setLinkedProductIds ] = useProductEntityProp<
+		number[]
+	>( property, { postType } );
+
+	useEffect( () => {
+		if ( ! state.selectedProduct ) {
+			loadLinkedProductsDispatcher( linkedProductIds ?? [] );
```

Comment:
Yes 2 things here:
1. The `useProductEntityProp` returns a nullish type even if we specify it. Like: 
```typescript
const [ value, setValue ] = useProductEntityProp<number[]>( property, { postType } );
```
`value: number[] | undefined` instead of `value: number[]` which I assumed that the `undefined` portion was covered by the `fallbackValue` of the `config` object.

2. The `useProductEntityProp` returns a nullish type even if we specify the `fallbackValue` as a `config`. Like: 
```typescript
const [ value, setValue ] = useProductEntityProp<number[]>( property, { postType, fallbackValue: [] } );
```
If the point 1 still returns a nullish type then the point 2 should not because we fallback always to a given value.